martinLi/assignments/reverseLinked/lc206/lc206.js

```diff
@@ -0,0 +1,86 @@
+// Problem Statement #
+
+// Given the head of a Singly LinkedList, reverse the LinkedList. Write a function to return the new head of the reversed LinkedList.
+
+class Node {
+  constructor(value, next = null) {
+    this.value = value;
+    this.next = next;
+  }
+
+  get_list() {
+    let result = "";
+    let temp = this;
+    while (temp !== null) {
+      result += temp.value + " ";
+      temp = temp.next;
+    }
+    return result;
+  }
+}
+
+const reverse = function (head) {
+  let prev = null;
+  /**
+   * [2] -> [4] -> [6] -> [8] -> [10] -> null
+   *                                    p
+   *  c
+   *         n
+   *  p
+   *         c
+   *                n
+   *
+   */
+  let current = head;
+  while (current != null) {
+    /**
+     * while current node isnt null
+     * we set next to the current next
+     * and we set the current next to prev
+     * current's next is pointed to the node previously
+     *
+     * current is set to next until we hit null
+     *
+     * We then return previous
+     */
```

Comment:
visualization is awsome, BUT, would probably say to put the viz and high level overview after the problem statement and before the program code begins